src/javascript/static/pages/home.js

```diff
@@ -24,11 +26,26 @@ const Home = (() => {
             // we need to initiate selector after it becoming visible
             TabSelector.repositionSelector();
 
-            const form_id = '#frm_verify_email';
-            FormManager.init(form_id, [
+            const utm_data = TrafficSource.getData();
+            const req = [
                 { selector: '#email', validations: ['req', 'email'], request_field: 'verify_email' },
                 { request_field: 'type', value: 'account_opening' },
-            ]);
+            ];
+
+            const url_parameters_req = {
+                request_field: 'url_parameters',
+                value        : { utm_source: TrafficSource.getSource(utm_data) },
+            };
+
+            if (utm_data.utm_campaign) url_parameters_req.value.utm_medium = utm_data.utm_medium;
+            if (utm_data.utm_campaign) url_parameters_req.value.utm_campaign = utm_data.utm_campaign;
+            const affiliate_token = Cookies.getJSON('affiliate_tracking');
+            if (affiliate_token) url_parameters_req.value.affiliate_token = affiliate_token.t;
+
+            req.push(url_parameters_req);
```

Comment:
same here. better yet, create a common function that gives back an object of url_parameters_req so we don't have to repeat it.

Reply:
Where do you think I should put that function?

Reply:
IMO in traffic_source would be fine

Reply:
Also since the entire `req` is similar, we could just use that and name it something like `getVerifyEmailFormRequest`

Reply:
`const getVerifyEmailFormRequest = () => {
    const utm_data = TrafficSource.getData();
    const affiliate_token = Cookies.getJSON('affiliate_tracking');

    return [
        { selector: '#email', validations: ['req', 'email'], request_field: 'verify_email' },
        { request_field: 'type', value: 'account_opening' },
        {
            request_field: 'url_parameters',
            value        : {
                utm_source: TrafficSource.getSource(utm_data),
                ...(utm_data.utm_campaign && {
                    utm_medium  : utm_data.utm_medium,
                    utm_campaign: utm_data.utm_campaign,
                }),
                ...(affiliate_token && { affiliate_token: affiliate_token.t }),
            },
        },
    ];
};`

I have this, I don't think we could put it there since it isn't fully related to traffic source but rather verify_email